### DISCUSSION TITLE
DaisyUI can't find type module

### DISCUSSION THREAD

**Discussion by huskyjp** [saadeghi/daisyui#819] [Q&A]
I do have Nextjs & Tailwind codebase and just implemented daisyui by `npm i daisyui` .
Then added `plugins: [require('daisyui')]` in tailwind.config.js file.

I get a warning error that says like below and I believe typescript currently can not find daisyui type file?? 

```
Could not find a declaration file for module 'daisyui'. '/Users/dev//node_modules/daisyui/src/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/daisyui` if it exists or add a new declaration (.d.ts) file containing `declare module 'daisyui';`ts(7016)
```

Also - when I tried to simply implemet some daisyui component like
```<button class="btn">My DaisyUI button</button>```
it just shows text without any daisyui design. 

I am not sure if the root problem is typescript error but when I start running the app, I get the daisyUI message so I believe the implementation is good but not working correctly for some reasaons.
```
 daisyUI components 2.15.1  https://github.com/saadeghi/daisyui
  ✔︎ Including:  base, components, themes[29], utilities
```

Would love to know following points 🙏
- the type error reason
- any possible reason why daisyui component is not working

**Comment by saadeghi** [comment]
@huskyjp for type error, see this:
https://stackoverflow.com/questions/46608937/npm-package-doesnt-have-types/46614891#46614891

About the style not showing up, can you please share an example repo so I can reproduce the issue?

**Reply by AliQ80** [reply]
I have the same problem

added the global.d.ts file and that fixed the first issue but the styles are still not showing
please check the fix-daisyUI branch in my repo

https://github.com/AliQ80/tutorly/tree/fix-daisyUI

daisyUI works fine if added through CDN as you can see I commented it in the index.html

thank you in advance

**Reply by shufolislam1** [reply]
did you find the solution? i have the same issue. working fine with cdn but having problem without cdn.

**Reply by meutiti** [reply]
Same issue, it doesn't work with next or whatever

**Reply by wushuanger** [reply]
works for me:
Create globals.d.ts in the root directory and add this line
`declare module "daisyui";`

**Comment by habdams** [comment]
A fix for this is to add a path to the daisyui on your node_modules folder, picture below show it:
![Screenshot from 2022-12-24 18-40-36](https://user-images.githubusercontent.com/6120474/209446393-05bf2143-54ab-486f-bdd4-c1f5255551fd.png)

**Reply by apischiang** [reply]
Thankyou! 🙏

**Comment by yash1031** [comment]
Adding global.d.ts file in root directory of projecting and writing -  declare module "daisyui"; into the file is resolving the problem successfully.
But still daisyUI styles are not working, so instead of installing daisyUI as a Tailwind plugin, I've used CDN for daisyUI which is importing the styles as expected.

Hope it helps to some extent.

Pls let me know for further explanation.

Thanks.

**Reply by saadeghi** [reply]
This topic is from 3 years ago.  
Later on types were added.  
And last month daisyUI 5 was released which works by adding it to CSS file like `@plugin "daisyui";` (not a JS import anymore)
